### ISSUE TITLE
Explore additional layout control surfaces for discoverability

### ISSUE THREAD

**Issue by sbatten** [microsoft/vscode#137633]
The goal of this issue is to experiment with new layout control surfaces to help users understand/discover the different layout customization options.

The initial version of this is available in insiders and disable-able via `workbench.experimental.layoutControl.enabled`

![image](https://user-images.githubusercontent.com/6561887/142807740-733a3040-2264-42d3-af47-4516a8205297.png)

**Comment by sbatten** [comment]
The latest here is 
<img width="241" alt="image" src="https://user-images.githubusercontent.com/6561887/150581490-834bda5e-2879-4544-a76b-ea0de2b21553.png">

<img width="563" alt="image" src="https://user-images.githubusercontent.com/6561887/150581527-5ff06ed7-819b-4973-8ec7-00293d1e5f85.png">

**Comment by heartacker** [comment]
hei, how can we exit the view without keyboard? 
![image](https://user-images.githubusercontent.com/1876302/150778809-79d2c332-2207-4e25-a47b-a90d4bfb0815.png)

I think we should add a close button to the left-top

**Comment by bpasero** [comment]
From testing: Some entries toggle on click, other entries are more like radio boxes, I think this picker needs a UX revisit.

**Comment by Tyriar** [comment]
My thoughts:

- I like the theme picker-like quick pick approach
- Can we get radio buttons supported?
- Can we use checkboxes? Instead of the eye?
- Highlighting sections based on the hovered/focused item is a great idea

**Comment by TylerLeonhardt** [comment]
> * Can we get radio buttons supported?
> * Can we use checkboxes? Instead of the eye?

Right now the quickpick is all or nothing with checkboxes so that would need to change but is a pretty large item.
Only after that would it be worth implementing radio button support for a group for this feature.

Also I might also say that having #120866 would be a dependency as I think a bunch of checkboxes (6 in this case) is kinda jarring over the approach @sbatten already took.

**Comment by misolori** [comment]
I think one of the way to fix the mixture of toggles is to switch to using checks for all items, which makes it easier to understand and it maps to the appearance menu. I'd also remove the layout modes as those aren't really customization. I'd also rename the titles to make them more readable:

<img width="638" alt="CleanShot 2022-01-25 at 14 46 24@2x" src="https://user-images.githubusercontent.com/35271042/151072574-0026959d-f440-4f29-845a-285cfc8a5c4b.png">

**Comment by sbatten** [comment]
@misolori this is not really possible with today's quick pick, but I agree on having a more uniform visualization. The current design spawns from another desire: I would like to come up with a UX that does not prevent us from adding custom visualization icons as well. On the left it would be nice to have codicons for each part to help the user identify the parts before selection

**Comment by misolori** [comment]
Based on @sbatten's idea on adding icons for each area to identify the area I made a little mockup that could show this:

<img width="646" alt="CleanShot 2022-01-25 at 16 31 15@2x" src="https://user-images.githubusercontent.com/35271042/151082660-6fe893c6-77df-4d7c-b3a9-4589aeb539f5.png">

There were also some other ideas from the design team around using the right controls (checkbox vs toggle), which would require some api changes, that i think properly represent each control:

<img width="633" alt="CleanShot 2022-01-25 at 16 33 15@2x" src="https://user-images.githubusercontent.com/35271042/151082866-1bf56ca3-d0ef-4a55-a06c-2a10d7757f98.png">

There's also the "simpler" approach of no icons:

<img width="637" alt="CleanShot 2022-01-25 at 16 34 02@2x" src="https://user-images.githubusercontent.com/35271042/151082909-1f526ef0-1265-4640-8945-7ac92c63420c.png">

**Comment by misolori** [comment]
Also +1 to highlighting the active area for the visibility items:

<img width="1092" alt="CleanShot 2022-01-25 at 17 15 53@2x" src="https://user-images.githubusercontent.com/35271042/151086750-fb6a6fb1-3afe-45c5-97fd-1859ce667aac.png">

**Comment by Tyriar** [comment]
@misolori

> I think one of the way to fix the mixture of toggles is to switch to using checks for all items, which makes it easier to understand and it maps to the appearance menu.

All checks feels bad to me since some are checks and some are radio buttons.

That last mockup looks fantastic though 👍

**Comment by gjsjohnmurray** [comment]
Short-term, how about prefixing mutually-exclusive labels with circle-filled or circle-outline, and toggles with check or nothing? Put the appropriate layout icon (after) that state-icon.

**Comment by gjsjohnmurray** [comment]
> Short-term, how about prefixing mutually-exclusive labels with circle-filled<strike> or circle-outline</strike>, and toggles with check or nothing? Put the appropriate layout icon (after) that state-icon.

A mockup I created using Developer Tools

![image](https://user-images.githubusercontent.com/6726799/151525293-34993071-6653-44f9-a30b-ee68c1e0718a.png)

Prefixes are \eab2 (check), \ea71 (circle-filled) and \2003 (EM SPACE)

**Comment by heartacker** [comment]
![image](https://user-images.githubusercontent.com/1876302/151588156-0214aec9-fa32-4dc0-a419-8cc6f9d0b2d5.png)
 maybe it's better to use dot instead for activity-bar and status-bar
I think it is hard to distinguish sidebar and activity-bar at first-quick peek

**Comment by GammaGames** [comment]
It's a really useful feature, especially including the menu bar since pressing alt frequently doesn't work.
How will users using the native title bar find the feature, though? If I didn't skim the release notes I wouldn't have found it.

**Comment by ElanHasson** [comment]
<img width="149" alt="image" src="https://user-images.githubusercontent.com/234704/152452834-e5a76e7f-c1e5-4f2c-a611-9309f7c9d825.png">
I'm not seeing that layout button.

**Comment by sbatten** [comment]
@ElanHasson You can add it with 
<img width="422" alt="image" src="https://user-images.githubusercontent.com/6561887/152454661-d5fb8a70-b530-4d08-9570-f3996160ed22.png">

**Comment by gjsjohnmurray** [comment]
@sbatten maybe link that description's mention of custom title bar to the setting that controls this.

**Comment by lexiconium** [comment]
I used to locate my terminal at the left side(right next to the side bar) like what "move panel to left" did in the previous version (1.63).
So I wanted to make it similar using this updated feature but I couldn't find the option that moves the side panel to the left side.
Or am I missing something?

**Comment by gjsjohnmurray** [comment]
@lexiconium see https://github.com/microsoft/vscode/issues/142075 https://github.com/microsoft/vscode/issues/141569

**Comment by adam-arold** [comment]
@ElanHasson @bpasero @heartacker @Tyriar @TylerLeonhardt please fix what you broke ( #141569 ) before experimenting with stuff.

**Comment by gjsjohnmurray** [comment]
@adam-arold what makes you think I had anything to do with this?

**Comment by adam-arold** [comment]
> @adam-arold what makes you think I had anything to do with this?

This issue was referenced in the release notes in the section that introduced the breaking change.

**Comment by AlbertoFabbri93** [comment]
I would like to have a quicker way to show/hide the side panel, something similar to the activity bar would be perfect.

**Comment by adam-arold** [comment]
> This issue was opened by (and is assigned to) @sbatten. I am not a member of the project team.

I see. Who can I talk to in this case if the issue was purposefully miscategorized as a feature request instead of a breaking change?

**Comment by gjsjohnmurray** [comment]
> I would like to have a quicker way to show/hide the side panel, something similar to the activity bar would be perfect.

@AlbertoFabbri93 are you seeing the Configure Layout button on your title bar?

![image](https://user-images.githubusercontent.com/6726799/152528188-e40a6790-f7d8-4cb5-9381-f3b7041de009.png)

If not, set `"workbench.experimental.layoutControl.enabled": true` and `"window.titleBarStyle": "custom"`

Hold down your mouse button on that button, drag the highlight down the menu to `Show Side Panel` item, then release the mouse button.

If you prefer a keyboard shortcut, bind one to the `View: Toggle Side Panel` command.

**Comment by ElanHasson** [comment]
> @ElanHasson You can add it with <img alt="image" width="422" src="https://user-images.githubusercontent.com/6561887/152454661-d5fb8a70-b530-4d08-9570-f3996160ed22.png">

Thanks @sbatten.

I'm also seeing an issue when I set a layout it doesn't persist across sessions. My expectation was that if I set a layout, it persists without additional config.

**Comment by sbatten** [comment]
@ElanHasson can you please file a separate issue with repro steps so I can confirm whether the behavior you are seeing is a bug?

**Comment by ElanHasson** [comment]
Will do.

**Comment by RedMustard** [comment]
Similar use case as @lexiconium. I'm curious if we're overlooking something. 

Prior to the new layout controls, I was able to dock my terminal side-by-side with the Side Bar on the right side of the window. As far as I can tell, the new layout controls prevent this type of layout and, instead, enforce the Side Panel to be on the opposite side of the window as the Side Bar.

**Comment by nickmccurdy** [comment]
The button doesn't appear in full screen on macOS, could we move it to the tab bar instead? I'd like to use this, but I almost always have VSCode full screen, so I can't without changing my workflow.

**Comment by AlbertoFabbri93** [comment]
@gjsjohnmurray Thank you very much for the reply, I have the Configure Layout button enabled but I feel like it is not that convenient if you are often toggling the visibility of the side panel.

**Comment by apple4ever** [comment]
I like the new feature, but it needs a quick access button (more than the button up top which requires two clicks, as opposed to the sidebar which only requires one click) and a key code (command-shift-B makes sense on a Mac).

**Comment by tacc-wbomar** [comment]
Please help me see the button. I do not see the button, in full screen nor windowed.

<details open>
<summary>My Settings:</summary>

- `window.titleBarStyle`: `custom`
- `workbench.experimental.layoutControl.enabled`: `true`
- `workbench.enableExperiments`: `true`

</details>

<details>
<summary>Steps Performed (click to toggle visibility):</summary>

0. In full-screen mode.
1. Confirm `window.titleBarStyle` setting is `custom` (was already set to this).
2. Change `workbench.experimental.layoutControl.enabled` to `true`.
3. From full-screen to windowed mode.
4. Restart VS Code.
5. In windowed mode.
6. Merge all windows.
7. From windowed to full-screen mode.
8. Confirm `workbench.enableExperiments` is set to `true` (was already set to this).
9. Cry here.

</details>

_VS Code 1.64.0 / MacOS Monterey 12.1_

**Comment by heartacker** [comment]
try my extension. 
https://marketplace.visualstudio.com/items?itemName=heartacker.toggle-panel
![](https://github.com/heartacker/toggle-panel/blob/master/images/toggle-panel.gif)

**Comment by IdemIsDito** [comment]
I'm not able to get my preferred panel setup with the new sidebar, previously I had:

<img width="2032" alt="image" src="https://user-images.githubusercontent.com/973821/153931088-78168f2a-2de5-4326-867f-e7cd1a2ba10a.png">

|------------editor------------|------side panel------|------side bar------|

**Comment by sbatten** [comment]
@IdemIsDito , the new side panel cannot be moved to the same side as the side bar. To do this, we have re-enabled you to move the bottom panel to the the right side with the side bar.

**Comment by sbatten** [comment]
For those following this issue, please try out this new setting in insiders: `workbench.experimental.layoutControl.type`

**Comment by heartacker** [comment]
wonderful

**Comment by heartacker** [comment]
![image](https://user-images.githubusercontent.com/1876302/155150057-5cdf5136-0f37-4912-8576-79d665036a45.png)
 inharmonious  menu icon

too bigger

**Comment by sbatten** [comment]
@misolori maybe we go back to the less abstract icon for workbench

**Comment by misolori** [comment]
we've updated the layout icon to be a bit smaller, open to more feedback:

<img width="233" alt="CleanShot 2022-02-25 at 09 31 43@2x" src="https://user-images.githubusercontent.com/35271042/155760611-e958d05b-ca66-4715-8270-00b75de73562.png">
